iceoryx_posh/test/moduletests/test_popo_server_port_user.cpp

```diff
@@ -0,0 +1,836 @@
+    auto errorHandlerGuard = iox::ErrorHandler::setTemporaryErrorHandler(
+        [&](const iox::Error error, const std::function<void()>, const iox::ErrorLevel errorLevel) {
+            EXPECT_THAT(error, Eq(iox::Error::kPOPO__CHUNK_SENDER_INVALID_CHUNK_TO_FREE_FROM_USER));
+            EXPECT_THAT(errorLevel, Eq(iox::ErrorLevel::SEVERE));
+            detectedError.emplace(error);
+        });
+
+    allocateResponseWithRequestHeaderAndThen(sut, [&](const auto, auto res) {
+        sut.portUser.freeResponse(res);
+        // since the response is already freed, it should not be in the UsedChunkList anymore and the error handler
+        // should be called
+        sut.portUser.freeResponse(res);
+    });
+
+    EXPECT_TRUE(detectedError.has_value());
+}
+
+TEST_F(ServerPort_test, FreeResponseWithWithNullptrResponseHeaderCallsTheErrorHandler)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "ecb40c4d-7b95-4780-9b51-ac1708830453");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    iox::cxx::optional<iox::Error> detectedError;
+    auto errorHandlerGuard = iox::ErrorHandler::setTemporaryErrorHandler(
+        [&](const iox::Error error, const std::function<void()>, const iox::ErrorLevel errorLevel) {
+            EXPECT_THAT(error, Eq(iox::Error::kPOPO__SERVER_PORT_INVALID_RESPONSE_TO_FREE_FROM_USER));
+            EXPECT_THAT(errorLevel, Eq(iox::ErrorLevel::SEVERE));
+            detectedError.emplace(error);
+        });
+
+    sut.portUser.freeResponse(nullptr);
+
+    EXPECT_TRUE(detectedError.has_value());
+}
+
+// END freeResponse tests
+
+// BEGIN sendResponse tests
+
+TEST_F(ServerPort_test, SendResponseWithWithNullptrResponseHeaderCallsTheErrorHandler)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "8d0b0a7f-d1ab-4249-b885-cbc6429eab83");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    iox::cxx::optional<iox::Error> detectedError;
+    auto errorHandlerGuard = iox::ErrorHandler::setTemporaryErrorHandler(
+        [&](const iox::Error error, const std::function<void()>, const iox::ErrorLevel errorLevel) {
+            EXPECT_THAT(error, Eq(iox::Error::kPOPO__SERVER_PORT_INVALID_RESPONSE_TO_SEND_FROM_USER));
+            EXPECT_THAT(errorLevel, Eq(iox::ErrorLevel::SEVERE));
+            detectedError.emplace(error);
+        });
+
+    sut.portUser.sendResponse(nullptr);
+
+    EXPECT_TRUE(detectedError.has_value());
+}
+
+TEST_F(ServerPort_test, SendResponseWithWithoutOfferReleasesTheChunkToTheMempool)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "dc4e31b1-18bf-4c42-9084-dd7abd52609b");
+    auto& sut = serverPortWithoutOfferOnCreate;
+
+    allocateResponseWithRequestHeaderAndThen(sut, [&](const auto, auto res) {
+        constexpr uint64_t NUMBER_OF_REQUEST_CHUNKS{1U};
+        constexpr uint64_t NUMBER_OF_RESPONSE_CHUNKS{1U};
+        EXPECT_THAT(this->getNumberOfUsedChunks(), Eq(NUMBER_OF_REQUEST_CHUNKS + NUMBER_OF_RESPONSE_CHUNKS));
+        sut.portUser.sendResponse(res);
```

Comment:
Can this even happen without the server having offered? I would think the server responds automatically when offered (and resources like memory chunks are sufficient). But I may be wrong here and testing that we have no leak either way makes sense anyway.